### ISSUE TITLE
How to support dbeaver for a new db? Tibero

### ISSUE THREAD

**Issue by luizplago** [dbeaver/dbeaver#6888]
Hey guys, I've been using DBeaver for a while and I really like it, the community is amazing and the developments it has had are fantastic.
I use it for everything, oracle, mysql, sql server ... but recently the company I work with decided to get a new db to replace oracle, this db is called Tibero I have used dbeaver with odbc to connect to this database.
I wonder how it would work so that dbeaver could support this Tibero.

Thank you so much for your attention.

**Comment by serge-rider** [comment]
If there is a JDBC driver for Tibero then you could try to add it by this instruction: https://github.com/dbeaver/dbeaver/wiki/Database-drivers

**Comment by luizplago** [comment]
> If there is a JDBC driver for Tibero then you could try to add it by this instruction: https://github.com/dbeaver/dbeaver/wiki/Database-drivers

Yes tibero has JDBC drive and works using the "Generic driver", but this generic has no sequences and some other features that you have when using the correct drive in dbeaver such as oracle.

As I said the way I've been using it is generic as if it were an ODBC driver

**Comment by serge-rider** [comment]
I see. Well, for extra functions (like sequences) we need to create a new extension. 
Unfortunately I'm not familiar with Tibero and we don't have any test Tibero server to validate our code.
You could try to contribute the extension (it is quite trivial for simple improvements like sequences, view/procedure source, etc). You can check how other similar extensions were implemented (e.g. H2).

Otherwise I can't say when we'll implement this. In any case we'll need more requests and assistance with Tibero installation.

**Comment by danimoya** [comment]
Also interested in this topic. In case you need a Tibero Server on the Cloud, I can provide you one. Thanks!

**Comment by ivankovalenko** [comment]
I've managed to create separate driver for Tibero and establish a connection but found some nuances.
There is a table with column A detected NUMBER by dbeaver. In Query column A/3 is returned as NUMBER(38,0) which does inproper round of returned value. Can this behavior be tuned in driver properties?

**Comment by ivankovalenko** [comment]
As for assistance for installing tibero - tmaxsoft (Tibero DB) does provide free trial vm with Tibero Database https://www.tmaxsoft.com/tibero-vm-download/. Hope that can move adoption Dbeaver for Tibero easier.

**Comment by younjunsu** [comment]
Hello,

Regarding the feature request I made previously in https://github.com/dbeaver/dbeaver/issues/15192, it was not possible to implement at the time due to the lack of an official driver link. However, TmaxTibero has recently provided one.

The Tibero database JDBC driver is now registered in the Maven Repository:
(https://mvnrepository.com/artifact/com.tmaxtibero/)

I believe it should now be possible to implement this feature using the official link above.

Could you please review this again?

Thank you.

**Comment by lhoris** [comment]
Hello.

I'm currently working on a Tibero database extension for DBeaver CE based on the Oracle plugin architecture.

The implementation already includes support for:

* Connection management
* Schema, table and view metadata
* Indexes, constraints and foreign keys
* Sequences
* Procedures and functions
* Packages and source editors
* DDL extraction
* Trigger support (under validation)

The goal is to provide a native Tibero experience similar to the existing Oracle extension.

I plan to submit a pull request once the implementation reaches a stable state.

Related to #6888.
